### PR TITLE
Replace Plugin-Local with Composite Build

### PR DESCRIPTION
**Pull request by DanielJette**
### What does this change accomplish?

Resolves https://github.com/ndtp/android-testify/issues/209

### How have you achieved it?

- https://docs.gradle.org/current/samples/sample_composite_builds_plugin_development.html#buildscript_dependencies_are_substituted
- https://docs.gradle.org/current/userguide/composite_builds.html#:~:text=In%20a%20composite%20build%2C%20Gradle,project%20path%20of%20the%20project.

### Scope of Impact and Testing instructions

The samples and libraries should be unaffected from this change. API should be consistent with 3.1.0, but the local builds will now use Composite Builds instead of requiring a local copy of the plugin jar.

I have tested and published an alpha version of this to [`3.2.0-alpha05`](https://repo1.maven.org/maven2/dev/testify/dev.testify.gradle.plugin/3.2.0-alpha05/)

### Notice

> [!WARNING]
> This change must keep `main` in a shippable state; **it may be shipped without further notice**.

### CODE REVIEW

**Comment by DanielJette** [comment]
```
> Task :Plugin:publishPluginMavenPublicationToMavenRepository
Multiple publications with coordinates 'dev.testify:dev.testify.gradle.plugin:3.2.0-alpha01' are published to repository 'maven'. The publications 'maven' in project ':Plugin' and 'pluginMaven' in project ':Plugin' will overwrite each other!

> Task :Plugin:generatePomFileForSimplePluginPluginMarkerMavenPublication

> Task :Plugin:publishSimplePluginPluginMarkerMavenPublicationToMavenRepository
Multiple publications with coordinates 'dev.testify:dev.testify.gradle.plugin:3.2.0-alpha01' are published to repository 'maven'. The publications 'maven' in project ':Plugin' and 'simplePluginPluginMarkerMaven' in project ':Plugin' will overwrite each other!
Multiple publications with coordinates 'dev.testify:dev.testify.gradle.plugin:3.2.0-alpha01' are published to repository 'maven'. The publications 'pluginMaven' in project ':Plugin' and 'simplePluginPluginMarkerMaven' in project ':Plugin' will overwrite each other!

> Task :Plugin:publish
```

**Comment by DanielJette** [comment]
**Event: Failed: POM Validation**
```
Invalid POM: /dev/testify/dev.testify.gradle.plugin/3.2.0-alpha01/dev.testify.gradle.plugin-3.2.0-alpha01.pom: Project name missing, Project description missing, Project URL missing, License information missing, SCM URL missing, Developer information missing
```

**Comment by DanielJette** [comment]
**Event: Failed: Signature Validation**
```
Invalid Signature: '/dev/testify/dev.testify.gradle.plugin/3.2.0-alpha01/dev.testify.gradle.plugin-3.2.0-alpha01.pom.asc' is not a valid signature for 'dev.testify.gradle.plugin-3.2.0-alpha01.pom'.
```

**Comment by DanielJette** [comment]
image

**Comment by DanielJette** [comment]
image
<br/>
image

**Comment by DanielJette** [comment]
The issue I keep running into is that composite builds appear to require the `java-gradle-plugin` plugin. However, this plugin manipulates the published artifacts in undesirable ways that I have not figured out how to control. It seems to duplicate and replace some of, but not all of, the publishing configuration we have set in `publish.build.gradle`

File: Plugins/Gradle/build.gradle [hunk 1]
```diff
@@ -22,11 +26,18 @@ ext {
...
 }
 
-version = "$project.versions.testify"
+version = gradleProperties.getProperty('testify_version')
 group = pom.publishedGroupId
-archivesBaseName = pom.artifact
 
-sourceCompatibility = JavaVersion.VERSION_17
+gradlePlugin {
+    automatedPublishing = false
```
Comment (jaredh) [inline Plugins/Gradle/build.gradle:33]: Probably a sign that we should move to letting the plugin generate the publications itself, but glad this is all it took.

**Review by jaredh — APPROVED** [review]
Love seeing that JAR deleted finally. 😍

File: Plugins/Gradle/build.gradle (hunk 1 above)
Comment (DanielJette) [reply Plugins/Gradle/build.gradle:33]: Absolutely not. Never touching publishing again unless something is deprecated or broken 😆 
I spent an embarrassing amount of time trying to get the built-in publishing to produce a properly signed plugin that was compatible with Maven Central. `gradlePlugin` is designed to publish to `https://plugins.gradle.org/` and the Maven Central support is confusing and seemingly impossible to configure properly.

**Pull request merged**